src/Microsoft.Data.Sqlite/Interop/VersionedMethods.cs

```diff
@@ -7,67 +7,56 @@ namespace Microsoft.Data.Sqlite.Interop
 {
     internal class VersionedMethods
     {
-        public static string SqliteErrorMessage(int rc, Sqlite3Handle db)
-        {
-            var message = db == null || db.IsInvalid
-                ? _strategy.ErrorString(rc)
-                : NativeMethods.sqlite3_errmsg(db);
+        private static readonly BaseStrategy _strategy;
 
-            return Strings.FormatSqliteNativeError(rc, message);
+        static VersionedMethods()
+        {
+            var version = new Version(NativeMethods.sqlite3_libversion());
+            _strategy = version >= new Version(3, 7, 15)
```

Comment:
Will revert. I got a little carried away with renames and simplifications. The if statements did look better.